### PR TITLE
Check for required PHP extensions

### PR DESCRIPTION
**Pull request by xh3n1**
## Related Issue

https://mantis.phplist.org/view.php?id=19837

### CODE REVIEW

File: public_html/lists/admin/index.php [hunk 1]
```diff
@@ -23,6 +23,42 @@
...
+
+$notinstalled = array();
+
+foreach ($phpExtensions as $value) {
+    if (!extension_loaded($value)) {
+        array_push($notinstalled, $value);
+    }
+}
+if (count($notinstalled) > 0) {
+    $message = "The following PHP extensions are missing:" . '<br>';
```
Comment (samtuke) [inline public_html/lists/admin/index.php]: Should this be translatable?

File: public_html/lists/admin/index.php [hunk 2]
```diff
@@ -23,6 +23,42 @@
 // Remove when php5.X is unsupported, currently 31 Dec 2018, https://secure.php.net/supported-versions.php
 require_once dirname(__FILE__).'/inc/random_compat/random.php';
 
+// Check if required extensions are installed.
+$phpExtensions = array(
```
Comment (samtuke) [inline public_html/lists/admin/index.php]: How about naming this $requiredExtensions for clarity?

File: public_html/lists/admin/index.php (hunk 1 above)
Comment (xh3n1) [reply public_html/lists/admin/index.php]: It cannot be translated as it stands above required language files.

File: public_html/lists/admin/index.php (hunk 2 above)
Comment (xh3n1) [reply public_html/lists/admin/index.php]: @samtuke OK

**Comment by bramley** [comment]
@michield @samtuke Despite what the documentation might say, some of the php extensions are not mandatory because the code sometimes checks before trying to use it.

Examples are curl and mbstring, see index.php
```
if (!function_exists('mb_strtolower')) {
    function mb_strtolower($string)
    {
        return strtolower($string);
    }
}

```

When this change is included in a release, if someone who does not have all of these extensions upgrades then phplist is going to stop working for them. I think that any change to making extensions mandatory needs a bit more thought.

**Comment by xh3n1** [comment]
@bramley 
There are not always checks for functions before trying to use them, which leads to blank pages etc.
Following your comment here, I created this PR https://github.com/phpList/phplist3/pull/513
where I moved the code to checkprerequisites.php, and removed some extensions that I think are more optional( those that have if-else step around functions).

**Pull request merged**